x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx

```diff
@@ -167,41 +169,53 @@ const IntendedRuleTypeEuiFormRow = styled(RuleTypeEuiFormRow)`
 
 // eslint-disable-next-line complexity
 const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
-  isLoading,
-  isUpdateView = false,
-  kibanaDataViews,
-  indicesConfig,
-  threatIndicesConfig,
+  browserFields,
+  dataSourceType,
   defaultSavedQuery,
+  enableThresholdSuppression,
   form,
-  optionsSelected,
-  setOptionsSelected,
+  groupByFields,
+  index,
   indexPattern,
+  indicesConfig,
   isIndexPatternLoading,
-  browserFields,
+  isLoading,
   isQueryBarValid,
+  isUpdateView = false,
+  kibanaDataViews,
+  optionsSelected,
+  queryBarSavedId,
+  queryBarTitle,
+  ruleType,
   setIsQueryBarValid,
   setIsThreatQueryBarValid,
-  ruleType,
-  index,
-  threatIndex,
-  groupByFields,
-  dataSourceType,
+  setOptionsSelected,
   shouldLoadQueryDynamically,
-  queryBarTitle,
-  queryBarSavedId,
+  threatIndex,
+  threatIndicesConfig,
   thresholdFields,
-  enableThresholdSuppression,
 }) => {
   const queryClient = useQueryClient();
 
   const { isSuppressionEnabled: isAlertSuppressionEnabled } = useAlertSuppression(ruleType);
-  const mlCapabilities = useMlCapabilities();
   const [openTimelineSearch, setOpenTimelineSearch] = useState(false);
   const [indexModified, setIndexModified] = useState(false);
   const [threatIndexModified, setThreatIndexModified] = useState(false);
   const license = useLicense();
 
+  const mlCapabilities = useMlCapabilities();
+  const [{ machineLearningJobId }] = useFormData<DefineStepRule>({
+    form,
+    watch: ['machineLearningJobId'],
+  });
+  const { someJobsStarted: someMlJobsStarted, noJobsStarted: noMlJobsStarted } =
+    useMlRuleValidations({ machineLearningJobId });
+  const { loading: mlFieldsLoading, fields: mlFields } = useRuleFields({ machineLearningJobId });
+  const mlSuppressionFields = useMemo(
+    () => getTermsAggregationFields(mlFields as BrowserField[]),
+    [mlFields]
+  );
+
```

Comment:
You're suggesting that this entire section be its own hook? What would you call it?

I already abstracted `useMlRuleValidations` and `useRuleFields` to address some of the complexity here, and `getTermsAggregationFields` is used elsewhere in this file so I opted not to include that as part of `useRuleFields`, but I'm open to suggestions.

Reply:
maybe `useMLRuleConfig` or `useMLRuleSetup`.

it might looks smth like that

```ts

const useMLruleSetup = (form: FromType) => {

  const mlCapabilities = useMlCapabilities();
    const [{ machineLearningJobId }] = useFormData<DefineStepRule>({
      form,
      watch: ['machineLearningJobId'],
    });
    const { someJobsStarted: someMlJobsStarted, noJobsStarted: noMlJobsStarted } =
      useMlRuleValidations({ machineLearningJobId });
    const { loading: mlFieldsLoading, fields: mlFields } = useRuleFields({ machineLearningJobId });
    const mlSuppressionFields = useMemo(
      () => getTermsAggregationFields(mlFields as BrowserField[]),
      [mlFields]
    );

  return {
mlSuppressionFields,
mlFieldsLoading,
someMlJobsStarted,
noMlJobsStarted,
}
}

```


Given, StepDeiunfeRule is already overloaded with the lot of logic, having this abstracted might be beneficial

Reply:
3aa105d0a53